### DISCUSSION TITLE
IFile LinkingUrl

### DISCUSSION THREAD

**Discussion by ChrTall** [pnp/pnpcore#698] [Q&A] [answered]
I have a questions about the LinkingUrl of IFiles. 
It seems that only Documents with the extensions: .docx, .xlsx, .pptx (probably all Office File Extensions)
are having a value for the LinkingUrl Property.
If I get a IFileCollection with .png or .pdf LinkingUrl is an empty string.
Is this a bug? I first thought it may be only possible to view Images and PDFs after downloading them from SharePoint, but when I click on the PDF or Image in SharePoint, it opens the Document like it does with Office Documents.

**Comment by jansenbe** [comment, marked as answer]
Hi @ChrTall ,

We don't do anything special here, what you see is what's returned by the SharePoint Online APIs and therefore by design.

**Reply by ChrTall** [reply]
@jansenbe Do you have a workaround for getting url of PDF Documents and Images? I am basically replacing a functionality that was done by using powershell scripts on the old On Premise Sharepoint. The Powershell script does not have this restriction to Office File Extensions. It also returns the Urls for PDF Files and others. Is this limitation only a limitation of ms graph or also CSOM? 
Thanks in advance for you great assistance.
